appinventor/blocklyeditor/src/blocks/components.js

```diff
@@ -251,8 +275,58 @@ Blockly.Blocks.component_event = {
   customContextMenu: function (options) {
     Blockly.FieldParameterFlydown.addHorizontalVerticalOption(this, options);
   },
+
```

Comment:
Every component block type has this verify function now. It is verified to see if the block is defined.
A block is verified only in two cases : 
1 : after it is loaded from its xml to block state. 
2 : after a component added or upgraded.